packages/beacon-node/src/monitoring/system.ts

```diff
@@ -5,264 +5,211 @@ import {Logger} from "@lodestar/utils";
-    collectNetworkData().catch((e) => debug("network", e)),
-  ]);
-
-  miscNodeBootTsSeconds = getSystemBootTime();
-}
+class System {
+  // static data only needs to be collected once
+  private staticDataCollected = false;
+  // disk I/O is not measurable in some environments
+  private diskIOMeasurable = true;
+
+  private _cpuCores = 0;
+  private _cpuThreads = 0;
+  private _cpuNodeSystemSecondsTotal = 0;
+  private _cpuNodeUserSecondsTotal = 0;
+  private _cpuNodeIdleSecondsTotal = 0;
+  private _memoryNodeBytesTotal = 0;
+  private _memoryNodeBytesFree = 0;
+  private _memoryNodeBytesCached = 0;
+  private _memoryNodeBytesBuffers = 0;
+  private _diskNodeBytesTotal = 0;
+  private _diskNodeBytesFree = 0;
+  private _diskNodeReadsTotal = 0;
+  private _diskNodeWritesTotal = 0;
+  private _networkNodeBytesTotalReceive = 0;
+  private _networkNodeBytesTotalTransmit = 0;
+  private _miscNodeBootTsSeconds = 0;
+  private _miscOs: MiscOs = "unk";
+
+  /**
+   * Collect system data and update cached values
+   */
+  async collectData(logger: Logger): Promise<void> {
+    const debug = (dataType: string, e: Error): void => logger.debug(`Failed to collect ${dataType} data`, {}, e);
+
+    await Promise.all([
+      this.collectStaticData().catch((e) => debug("static system", e)),
+      this.collectCpuData().catch((e) => debug("CPU", e)),
+      this.collectMemoryData().catch((e) => debug("memory", e)),
+      this.collectDiskData().catch((e) => debug("disk", e)),
+      this.collectNetworkData().catch((e) => debug("network", e)),
+    ]);
+
+    this._miscNodeBootTsSeconds = this.getSystemBootTime();
+  }
 
-async function collectStaticData(): Promise<void> {
-  if (staticDataCollected) return;
+  private async collectStaticData(): Promise<void> {
+    if (this.staticDataCollected) return;
 
-  const cpu = await system.cpu();
-  // Note: inside container this might be inaccurate as
-  // physicalCores in some cases is the count of logical CPU cores
-  cpuCores = cpu.physicalCores;
-  cpuThreads = cpu.cores;
+    const cpu = await system.cpu();
+    // Note: inside container this might be inaccurate as
+    // physicalCores in some cases is the count of logical CPU cores
+    this._cpuCores = cpu.physicalCores;
+    this._cpuThreads = cpu.cores;
 
-  miscOs = getNormalizedOsVersion();
+    this._miscOs = this.getNormalizedOsVersion();
 
-  staticDataCollected = true;
-}
+    this.staticDataCollected = true;
+  }
 
-async function collectCpuData(): Promise<void> {
-  const cpuTimes: Record<string, number> = {};
+  private async collectCpuData(): Promise<void> {
+    const cpuTimes: Record<string, number> = {};
 
-  os.cpus().forEach((cpu) => {
-    // sum up CPU times per mode and convert to seconds
-    for (const [mode, time] of Object.entries(cpu.times)) {
-      if (cpuTimes[mode] == null) cpuTimes[mode] = 0;
-      cpuTimes[mode] += Math.floor(time / 1000);
+    for (const cpu of os.cpus()) {
```

Comment:
changed this from `os.cpus().forEach((cpu) => {})` to `for (const cpu of os.cpus())`, see https://github.com/ChainSafe/lodestar/pull/5161#discussion_r1116132778